lib/hash.js

```diff
@@ -0,0 +1,74 @@
+const crypto = require('crypto');
+
+const SALT_LENGTH = 64;
+const HASH_ITERATIONS = 10000;
+const HASH_LENGTH = 256;
+const CIPHER = 'sha512';
+const SEPARATOR = ':';
+
+function genSalt() {
+  return crypto.randomBytes(SALT_LENGTH).toString('hex');
+}
+
+/**
+ * @callback saltAndHashCallback
+ * @param {Error}
+ * @param {String} responseMessage
+ */
+
+/**
+ * Generates a random salt and encrypts the given password
+ * @param  {String}   pwd Password to be encrypted
+ * @param  {saltAndHashCallback} cb Node-style callback
+ */
+exports.saltAndHash = function(pwd, cb) {
+  const salt = genSalt();
+  crypto.pbkdf2(pwd, salt, HASH_ITERATIONS, HASH_LENGTH, CIPHER, function(err, hashed) {
```

Comment:
Added support for exponential delay, see [this commit](https://github.com/feedhenry-raincatcher/raincatcher-demo-auth/pull/23/commits/4cb1080a4548c9dde7c9c479b463c425037938fd)